demo/api/src/products/generated/dto/products.args.ts

```diff
@@ -0,0 +1,25 @@
+import { OffsetBasedPaginationArgs } from "@comet/cms-api";
+import { ArgsType, Field } from "@nestjs/graphql";
+import { Type } from "class-transformer";
+import { IsOptional, IsString, ValidateNested } from "class-validator";
+
+import { ProductFilter } from "./product.filter";
+import { ProductSort } from "./product.sort";
+
+@ArgsType()
+export class ProductsArgs extends OffsetBasedPaginationArgs {
+    @Field({ nullable: true })
+    @IsOptional()
+    @IsString()
+    search?: string;
+
+    @Field(() => ProductFilter, { nullable: true })
```

Comment:
not related to this PR, but nullable fields should have @IsOptional for validation

Reply:
I've created a task for this 👍🏼

Reply:
https://github.com/vivid-planet/comet/pull/1064